tests/pass/concurrency/simple.rs

```diff
@@ -1,4 +1,5 @@
 // ignore-windows: Concurrency on Windows is not supported yet.
+// ignore-android: `libc` crate does not support `PR_SET_NAME` on Android
```

Comment:
This code does not directly call `libc` though. It just uses standard library APIs. So it should work on Android as well, right?

Reply:
Sorry, these comments in the tests were notes about the root cause of why the feature is not supported, and therefore why the test is ignored. I can reword this and the other locations to be more concise.

FWIW this is the (trimmed) panic you receive running this test:

```
thread '<unnamed>' panicked at 'Hello!', tests/pass/concurrency/simple.rs:55:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'rustc' panicked at 'failed to find required Rust item: ["libc", "PR_SET_NAME"]', src/helpers.rs:82:32
stack backtrace:
   0:        0x1027845f4 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h34a40067a35bfbec
   1:        0x1027def6b - core::fmt::write::h90d56a63b82c1d78
   2:        0x102776c18 - std::io::Write::write_fmt::h9a11e832e4a1e526
   3:        0x10278797d - std::panicking::default_hook::{{closure}}::h50f5199bcbe28750
   4:        0x1027876f7 - std::panicking::default_hook::h95b189cf93ee7eab
   5:        0x10fe9a05d - rustc_driver[8cbe49adf889e25c]::DEFAULT_HOOK::{closure#0}::{closure#0}
   6:        0x102787fdb - std::panicking::rust_panic_with_hook::hde5cf82e5f0b952b
   7:        0x102787e53 - std::panicking::begin_panic_handler::{{closure}}::h02042140a786e7a4
   8:        0x102784a77 - std::sys_common::backtrace::__rust_end_short_backtrace::ha59ad93f482bafe0
   9:        0x102787b2a - _rust_begin_unwind
  10:        0x102808703 - core::panicking::panic_fmt::hd46cbc4cb5821886
  11:        0x101e2404f - miri::helpers::EvalContextExt::resolve_path::{{closure}}::h0ec9ed0c1bb395ff
                               at /Users/hvx/dev/miri/src/helpers.rs:82:32
  12:        0x101e2404f - core::option::Option<T>::unwrap_or_else::h0d7d7e05cfeea290
                               at /rustc/e71440575c930dcecac288b7c3536410d688b351/library/core/src/option.rs:825:21
  13:        0x101e2404f - miri::helpers::EvalContextExt::resolve_path::h00886eb1a3d5bd0f
                               at /Users/hvx/dev/miri/src/helpers.rs:81:9
  14:        0x101e2404f - miri::helpers::EvalContextExt::eval_path_scalar::h8283fff130397323
                               at /Users/hvx/dev/miri/src/helpers.rs:89:24
  15:        0x101e65095 - miri::helpers::EvalContextExt::eval_libc::h5c2d4c27033a0621
                               at /Users/hvx/dev/miri/src/helpers.rs:98:9
  16:        0x101e65095 - miri::helpers::EvalContextExt::eval_libc_i32::h506af73185e8afd7
                               at /Users/hvx/dev/miri/src/helpers.rs:104:9
  17:        0x101e65095 - miri::shims::posix::thread::EvalContextExt::prctl::he65e14925b5442ba
                               at /Users/hvx/dev/miri/src/shims/posix/thread.rs:110:22
  18:        0x101e65fb4 - miri::shims::posix::linux::foreign_items::EvalContextExt::emulate_foreign_item_by_name::h2a3aca9b29812087
                               at /Users/hvx/dev/miri/src/shims/posix/linux/foreign_items.rs:110:30
  19:        0x101e65796 - miri::shims::posix::android::foreign_items::EvalContextExt::emulate_foreign_item_by_name::h2477024ea6b084a5
                               at /Users/hvx/dev/miri/src/shims/posix/android/foreign_items.rs:24:24
  20:        0x101e4fc1c - miri::shims::posix::foreign_items::EvalContextExt::emulate_foreign_item_by_name::h972a07cae6d0156c
                               at /Users/hvx/dev/miri/src/shims/posix/foreign_items.rs:499:41
  21:        0x101e2fb1c - miri::shims::foreign_items::EvalContextExt::emulate_foreign_item_by_name::hc7eb92b0e9675c1b
                               at /Users/hvx/dev/miri/src/shims/foreign_items.rs:705:63
  22:        0x101df5e85 - miri::shims::foreign_items::EvalContextExt::emulate_foreign_item::hf8d1dc31e7e38b74
                               at /Users/hvx/dev/miri/src/shims/foreign_items.rs:302:15
  23:        0x101df5e85 - miri::shims::EvalContextExt::find_mir_or_eval_fn::h67395dcac9bbe974
                               at /Users/hvx/dev/miri/src/shims/mod.rs:54:20
```

Reply:
Right, that is caused by you calling our linux-specific implementation of `prctl` on a non-Linux target.

This confirms my stance that android should not just fall back to the Linux functions. That leads to the weirdest failures, in the best case -- or to silently wrong behavior in the worst case.